Packages/Tracking/Examples~/Example Assets/3D Interaction/Turntable And Pullcord/Scripts/Turntable/Turntable.cs

```diff
@@ -0,0 +1,200 @@
+        private float _edgeLength;
+
+        [Range(0, 90)]
+        [SerializeField]
+        [Tooltip("The angle the edge forms with the upper section of the turntable.")]
+        private float _edgeAngle = 45;
+
+        [Header("Turntable Motion")]
+        [MinValue(0)]
+        [SerializeField]
+        [Tooltip("How much to scale the rotational motion by.  A value of 1 causes no extra scale.")]
+        private float _rotationScale = 1.5f;
+
+        [MinValue(0.00001f)]
+        [SerializeField]
+        [Tooltip("How much to smooth the velocity while the user is touching the turntable.")]
+        private float _rotationSmoothing = 0.1f;
+
+        [Range(0, 1)]
+        [SerializeField]
+        [Tooltip("The damping factor to use to damp the rotational velocity of the turntable.")]
+        private float _rotationDamping = 0.95f;
+
+        [MinValue(0)]
+        [SerializeField]
+        [Tooltip("The speed under which the turntable will stop completely.")]
+        private float _minimumSpeed = 0.01f;
+
+        public float LowerLevelHeight
+        {
+            get
+            {
+                return _tableHeight - _edgeLength * Mathf.Sin(_edgeAngle * Mathf.Deg2Rad);
+            }
+        }
+
+        public float LowerLevelRadius
+        {
+            get
+            {
+                return _tableRadius + _edgeLength * Mathf.Cos(_edgeAngle * Mathf.Deg2Rad);
+            }
+        }
+
+        // Maps a finger from a specific finger to the world tip position when it first entered the turntable
+        private Dictionary<int, Vector3> _currTipPoints = new Dictionary<int, Vector3>();
+        private Dictionary<int, Vector3> _prevTipPoints = new Dictionary<int, Vector3>();
+
+        private SmoothedFloat _smoothedVelocity;
+        private float _rotationalVelocity;
+
+        private TurntableVisuals _turntableVisuals;
+
+        private void OnValidate()
+        {
+            if (!Application.isPlaying)
+            {
+                if (_turntableVisuals == null) _turntableVisuals = GetComponent<TurntableVisuals>();
+                if (_turntableVisuals != null) _turntableVisuals.UpdateVisuals();
+            }
+        }
+
+        private void Awake()
+        {
+            if (_provider == null)
+            {
+                _provider = Hands.Provider;
+            }
+
+            _smoothedVelocity = new SmoothedFloat();
+            _smoothedVelocity.delay = _rotationSmoothing;
+
+            if (_turntableVisuals == null) _turntableVisuals = GetComponent<TurntableVisuals>();
+            if (_turntableVisuals != null) _turntableVisuals.ArcDegrees = 360;
+        }
+
+        private void Update()
+        {
+            Utils.Swap(ref _currTipPoints, ref _prevTipPoints);
+
+            _currTipPoints.Clear();
+            foreach (var hand in _provider.CurrentFrame.Hands)
+            {
+                int fingerID = 0;
```

Comment:
Does using the fingerID here work for two handed interaction with the turntable?
I assume this would completely ignore the left hand's fingertips, if the right hand's same fingertips are intersecting with the turntable at the same time.

Could this be changed to sth like 
```suggestion
                int fingerID = hand.isLeft ? 0 : hand.Fingers.Length;
```

Reply:
yeah that sounds great, we should add a comment explaining what that represents though as it's not overly readable :)